packages/pieces/stability-ai/src/lib/actions/text-to-image.ts

```diff
@@ -0,0 +1,280 @@
+            label: 'low-poly',
+            value: 'low-poly',
+          },
+          {
+            label: 'origami',
+            value: 'origami',
+          },
+          {
+            label: 'modeling-compound',
+            value: 'modeling-compound',
+          },
+          {
+            label: 'cinematic',
+            value: 'cinematic',
+          },
+          {
+            label: '3d-model',
+            value: '3d-model',
+          },
+          {
+            label: 'pixel-art',
+            value: 'pixel-art',
+          },
+          {
+            label: 'tile-texture',
+            value: 'tile-texture',
+          },
+        ],
+      },
+    }),
+    engine_id: Property.StaticDropdown({
+      displayName: 'Engine ID',
+      required: true,
+      options: {
+        options: [
+          {
+            label: 'stable-diffusion-768-v2-1',
+            value: 'stable-diffusion-768-v2-1',
+          },
+          {
+            label: 'stable-diffusion-512-v2-1',
+            value: 'stable-diffusion-512-v2-1',
+          },
+          {
+            label: 'stable-diffusion-768-v2-0',
+            value: 'stable-diffusion-768-v2-0',
+          },
+          {
+            label: 'stable-diffusion-512-v2-0',
+            value: 'stable-diffusion-512-v2-0',
+          },
+          {
+            label: 'stable-diffusion-v1-5',
+            value: 'stable-diffusion-v1-5',
+          },
+          {
+            label: 'stable-diffusion-v1',
+            value: 'stable-diffusion-v1',
+          },
+        ],
+      },
+    }),
+    accept_format: Property.StaticDropdown({
+      displayName: 'accept_format',
+      description:
+        'json will return the image as base64, and png will return the image as binary',
+      required: true,
+      options: {
+        options: [
+          {
+            label: 'application/json',
+            value: 'application/json',
+          },
+          {
+            label: 'image/png',
+            value: 'image/png',
+          },
+        ],
+      },
+    }),
+  },
+  async run(context) {
+    const {
+      prompt,
+      cfg_scale,
+      clip_guidance_preset,
+      height,
+      width,
+      samples,
+      steps,
+      accept_format,
+      style_preset,
+      engine_id,
+      weight,
+    } = context.propsValue;
+
+    const engineId = engine_id || 'stable-diffusion-v1-5';
+    const apiHost = 'https://api.stability.ai';
+
+    const apiKey = context.propsValue.authentication.api_key;
+
+    const requestBody = {
+      text_prompts: [
+        {
+          text: prompt,
+          weight: Number(weight) || 1,
+        },
+      ],
+      cfg_scale: Number(cfg_scale) || 7,
+      clip_guidance_preset: clip_guidance_preset || 'NONE',
+      height: Number(height) || 512,
+      width: Number(width) || 512,
+      samples: Number(samples) || 1,
+      steps: Number(steps) || 50,
+      style_preset,
+    };
+
+    try {
+      const response = await axios.post(
```

Comment:
can you use our http client, It's wrapper for axios and it also provide betters errors, so you don't need to catch it and reformat it.